### ISSUE TITLE
EntityManagerFactory "rowId is null"-error after upgrading to 3.1.5

### ISSUE THREAD

**Issue by NicoStrecker** [spring-projects/spring-boot#37972]
Hello i upgraded from Spring boot version 3.1.2 to 3.1.5 since then i get the following error:
```log
 - 2023-10-20 12:17:36.097 [main] ERROR  org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'entityManagerFactory' defined in class path resource [org/springframework/boot/autoconfigure/orm/jpa/HibernateJpaConfiguration.class]: Cannot invoke "String.isEmpty()" because "rowId" is null
```
I havent changed anything else

**Comment by wilkinsona** [comment]
This is likely to be a Hibernate problem. By upgrading from Spring Boot 3.1.2 to 3.1.5 you have also upgraded from Hibernate 6.2.6.Final to 6.2.13.Final. I recommend experimenting with different Hibernate versions to see if you can identify if and when the problem was introduced in Hibernate.

If that does not help and you would like us to spend some more time investigating, please spend some time providing a complete yet minimal sample that reproduces the problem. You can share it with us by pushing it to a separate repository on GitHub or by zipping it up and attaching it to this issue.

**Comment by NicoStrecker** [comment]
Downgrading to 3.1.4 works so i can narrow down the issue between 6.2.9 and 6.2.13.
After iterating through each dependency i found out the issue is in the Hibernate dependency  6.2.13.Final, Spring-Boot 3.1.4 works with all Hibernate Versions except 6.2.13.Final and so does Spring-Boot 3.1.5

I would love to provide a minimal sample but we use a Db2zDialect where the issue occurs (our H2 Tests work so far).
I will create a issue in a Hibernate Tracker and link it here

**Comment by NicoStrecker** [comment]
Hibernate-Issue: https://hibernate.atlassian.net/browse/HHH-17344

**Comment by wilkinsona** [comment]
Thanks. Please do open a Hibernate issue and comment here with a link to it so that other's can track the problem.

**Comment by NicoStrecker** [comment]
Problem is solved in hibernate core 6.4.0
